crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -643,7 +650,7 @@ impl Backend {
 
     /// Reverts the state to the snapshot identified by the given `id`.
     pub async fn revert_snapshot(&self, id: U256) -> Result<bool, BlockchainError> {
-        let block = { self.active_snapshots.lock().remove(&id) };
+        let block = { self.active_snapshots.lock().get(&id).cloned() };
```

Comment:
this is now a `revert_to`?

Reply:
strictly speaking, the rest of this function is fallible, so is it a good idea to clear the list before we actually succeed?